File: replace_imports.py
import os
import re

count = 0
for root, _, files in os.walk('./apps'):
    if 'node_modules' in root or '.next' in root:
        continue
    for f in files:
        if not (f.endswith('.ts') or f.endswith('.tsx')): continue
        path = os.path.join(root, f)
        with open(path, 'r') as file:
            content = file.read()
            
        pattern = r"from\s+['\"]([^'\"]*lib/types|[^'\"]*types)['\"]"
        if re.search(pattern, content):
            new_content = re.sub(pattern, r"from '@kloqo/shared'", content)
            if new_content != content:
                with open(path, 'w') as file:
                    file.write(new_content)
                print(f"Updated {path}")
                count += 1

print(f"Total files updated: {count}")

File: replace_imports.js
const fs = require('fs');
const path = require('path');

function walk(dir) {
    let results = [];
    const list = fs.readdirSync(dir);
    list.forEach(function(file) {
        file = dir + '/' + file;
        const stat = fs.statSync(file);
        if (stat && stat.isDirectory() && !file.includes('node_modules') && !file.includes('.next')) { 
            results = results.concat(walk(file));
        } else if (file.endsWith('.ts') || file.endsWith('.tsx') || file.endsWith('.js') || file.endsWith('.jsx')) {
            results.push(file);
        }
    });
    return results;
}

const files = walk('./apps');
const regex = /from\s+['"]([^'"]*lib\/types|[^'"]*\/types)['"]/g;

files.forEach(file => {
    let content = fs.readFileSync(file, 'utf8');
    let hasChanges = false;
    
    // Check if there are any matches
    if (regex.test(content)) {
        content = content.replace(regex, "from '@kloqo/shared'");
        fs.writeFileSync(file, content, 'utf8');
        console.log('Updated', file);
    }
});
